Support `pod.spec.containers.securityContext`  specification

LittleWat opened the pull request:
This PR attempts to close #116

LittleWat commented:
@hashhar sorry to ask you but could you kindly review this, please...? 🙏

chgl commented:
Not a maintainer, but I'm wondering if it makes sense to define a default, restrictive securityContext out-of-the-box. Trino should support running as non-root and without privilege (escalation) requirements. Most, if not all, of the bitnami charts do this, e.g. https://github.com/bitnami/charts/blob/main/bitnami/postgresql/values.yaml#L477.

LittleWat commented:
@chgl thank you for your comments! That would be better!

Update this PR following the [valeriano-manassero's implementation](https://github.com/valeriano-manassero/helm-charts/blob/6382a14272927a908bc006d0f1370ba9dffc821f/valeriano-manassero/trino/values.yaml#L467-L471)

@mosabua sorry to ask you but I fixed the commit and rebased this branch to the latest main branch. Could you review this when you have time, please...? 🙏

LittleWat commented on charts/trino/README.md:
```diff
@@ -49,6 +49,8 @@ The following table lists the configurable parameters of the Trino chart and the
 | `sidecarContainers` |  | `{}` |
 | `securityContext.runAsUser` |  | `1000` |
 | `securityContext.runAsGroup` |  | `1000` |
+| `containerSecurityContext.allowPrivilegeEscalation` | Controls whether a process can gain more privileges than its parent process. | `false` |
```
I generated this README using the command `frigate gen . > README.md ` so it would be okay

mosabua replied:
Thanks

LittleWat commented on charts/trino/README.md:
```diff
@@ -49,6 +49,8 @@ The following table lists the configurable parameters of the Trino chart and the
 | `securityContext.runAsUser` |  | `1000` |
 | `securityContext.runAsGroup` |  | `1000` |
+| `containerSecurityContext.allowPrivilegeEscalation` | Controls whether a process can gain more privileges than its parent process. | `false` |
+| `containerSecurityContext.capabilities.drop` | Kernel capabilities that will be dropped from a pod | `["ALL"]` |
```
updated the description following your advice!

LittleWat commented:
@mosabua thank you for your review! I have amended the commit to update the description! could you check this again, please...?

mosabua commented on charts/trino/values.yaml:
```diff
@@ -218,6 +218,13 @@ securityContext:
+containerSecurityContext:
+  allowPrivilegeEscalation: false  #  Controls whether a process can gain more privileges than its parent process.
+  capabilities:
+    drop:  # List of Linux kernel capabilities that are dropped from every container. You can confirm the options for "capabilities" here:　https://man7.org/linux/man-pages/man7/capabilities.7.html  Please make sure to remove "CAP_" prefix which the kernel attaches to the names of permissions.
```
Use

List of Linux kernel capabilities that are dropped from every container. Valid valeus are listed at https://man7.org/linux/man-pages/man7/capabilities.7.html  Ensure to remove the "CAP_" prefix which the kernel attaches to the names of permissions.

mosabua replied:
Can you adjust to this?

LittleWat replied:
oh, sorry. Applied your suggestion! Thank you!

mosabua commented on charts/trino/values.yaml:
```diff
@@ -218,6 +218,13 @@ securityContext:
 
+# -- SecurityContext configuration for containers
+containerSecurityContext:
+  allowPrivilegeEscalation: false  #  Controls whether a process can gain more privileges than its parent process.
```
```suggestion
  allowPrivilegeEscalation: false  #  Control whether a process can gain more privileges than its parent process.
```

LittleWat replied:
applied!

mosabua commented on charts/trino/values.yaml:
```diff
@@ -218,6 +218,13 @@ securityContext:
+  allowPrivilegeEscalation: false  #  Controls whether a process can gain more privileges than its parent process.
+  capabilities:
+    drop:  # List of Linux kernel capabilities that are dropped from every container. You can confirm the options for "capabilities" here:　https://man7.org/linux/man-pages/man7/capabilities.7.html  Please make sure to remove "CAP_" prefix which the kernel attaches to the names of permissions.
+      - ALL
```
Is this meaningful and good default value? Is it compatible with the current behavior?

LittleWat replied:
yes, because the Trino containers will not require any Linux kernel capabilities.
It would be a best practice to remove unnecessary capabilities from containers.
I've confirmed that this works fine in our environment.

mosabua replied:
Thanks for confirming

bond- commented:
How about keeping all of the security context open to values.yaml

```
          securityContext:
            {{- toYaml .Values.securityContext | nindent 12 }}
```

LittleWat commented:
@bond- Thank you for your comment! yes, that's exactly what I did in the first place. But I set the default following the comment in this review by amending the commit 🙇

nineinchnick commented:
@LittleWat please rebase, and I'll review this once the new tests in the CI pass.

LittleWat commented:
@nineinchnick thanks! rebased!

nineinchnick commented on charts/trino/values.yaml:
```diff
@@ -267,6 +267,13 @@ securityContext:
   runAsUser: 1000
   runAsGroup: 1000
 
+# -- SecurityContext configuration for containers
```
```suggestion
# -- [Container security context](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-container) configuration.
```

nineinchnick commented on charts/trino/values.yaml:
```diff
@@ -267,6 +267,13 @@ securityContext:
+containerSecurityContext:
+  allowPrivilegeEscalation: false  #  Control whether a process can gain more privileges than its parent process.
+  capabilities:
+    drop:  # List of Linux kernel capabilities that are dropped from every container. Valid values are listed at https://man7.org/linux/man-pages/man7/capabilities.7.html Ensure to remove the "CAP_" prefix which the kernel attaches to the names of permissions.
```
If you prefix it with `--`, it'll appear in the docs (the README). The comment is important, so I think it's worth doing.

```suggestion
    # -- A list of the Linux kernel capabilities that are dropped from every container. Valid values are listed at https://man7.org/linux/man-pages/man7/capabilities.7.html Ensure to remove the "CAP_" prefix which the kernel attaches to the names of permissions.
    drop:
```

nineinchnick reviewed (approved):
Just two nitpicks and it'll be good to go!

nineinchnick commented on charts/trino/values.yaml:
```diff
@@ -267,6 +267,13 @@ securityContext:
 
+# -- SecurityContext configuration for containers
+containerSecurityContext:
+  allowPrivilegeEscalation: false  #  Control whether a process can gain more privileges than its parent process.
```
```suggestion
  # -- Control whether a process can gain more privileges than its parent process.
  allowPrivilegeEscalation: false  
```

LittleWat commented:
@nineinchnick thank you for your quick review! yes, that would be better! Fixed!
